src/SmartComponents/FormComponents/DataListLayout.js

```diff
@@ -0,0 +1,83 @@
+import React, { Fragment, Component } from 'react';
+import {
+    DataList,
+    DataListItem,
+    DataListItemRow,
+    DataListItemCells,
+    DataListCell
+} from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+
+class DataListRow extends Component {
+    state = {
+        hasError: false
+    }
+
+    componentDidCatch(_error, info) {
+        this.setState({ hasError: info });
+    }
+
```

Comment:
You can have `shouldComponentUpdate` here to prevent re-renders on schema changes. (or memo if this was function).

Child components should have their own triggers based on form state changes.